src/Magento/FunctionalTestingFramework/Allure/Adapter/MagentoAllureAdapter.php

```diff
@@ -220,4 +233,20 @@ function () use ($rootStep, $formattedSteps) {
 
         $this->getLifecycle()->fire(new TestCaseFinishedEvent());
     }
+
+    private function retrieveStepKey($stepLine)
+    {
+        $stepKey = null;
+        list($filePath, $stepLine) = explode(":", $stepLine);
+        $prevStepLine = $stepLine - 2;
+
+        if (!array_key_exists($filePath, $this->testFiles)) {
+            $this->testFiles[$filePath] = explode(PHP_EOL, file_get_contents($filePath));
+        }
+        $testFile = $this->testFiles[$filePath];
+
+        list($stepKey) = sscanf($testFile[$prevStepLine], TestGenerator::STEP_KEY_ANNOTATION);
```

Comment:
The thing is it will not help preventing weird behavior.
Cause for `comment` it returns line number 44 and it will find stepKey from another action.
And if into `sscanf` first arg will be sent as null it will return null. This is ok for us I think.

Reply:
Oh gotcha. Index out of bounds for `array[bad index]` is only a notice so it returns null anyways.
In that case I agree with ya.